AsyncNats/Util/Utf8String.cs

```diff
@@ -0,0 +1,154 @@
+﻿
+
+namespace EightyDecibel.AsyncNats
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Runtime.CompilerServices;
+    using System.Text;
+
+
+    public readonly struct NatsKey : IEquatable<NatsKey>, IEquatable<string>
```

Comment:
What's the added benefit of this function? Making InternalSubscribe slightly less complex? Or to supply the 'Unsafe' version? [which I'd like to see removed, not a huge fan of 'Unsafe']